modules/dists/BlockCycDist.chpl

```diff
@@ -482,24 +458,30 @@ class LocBlockCyclic {
   proc init(param rank: int,
             type idxType,
             locid,   // the locale index from the target domain
-            dist: unmanaged BlockCyclic(rank, idxType)) { // reference to glob dist
+            lowIdx, blocksize, targetLocDom) {  // from global dist
```

Comment:
I am now passing pieces of the global distribution class to the initializer of the local class, instead of passing the distribution class itself to the initializer. This change is needed because now the local classes are created first, are packaged into the array locDistTemp, then passed to the initializer of the global distribution class. So the global distribution class does not yet exist when the local classes are created, so I can’t pass it down.